src/components/StampOverlay/StampOverlay.js

```diff
@@ -0,0 +1,169 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { withTranslation } from 'react-i18next';
+import onClickOutside from 'react-onclickoutside';
+import getClassName from 'helpers/getClassName';
+import getToolStylePopupPositionBasedOn from 'helpers/getToolStylePopupPositionBasedOn';
+import actions from 'actions';
+import selectors from 'selectors';
+import core from 'core';
+
+import './StampOverlay.scss';
+
+const TOOL_NAME = 'AnnotationCreateRubberStamp';
+const canvasWidth = 160;
+const canvasHeight = 58;
+
+
+class StampOverlay extends React.Component {
+  static propTypes = {
+    activeToolName: PropTypes.string,
+    isDisabled: PropTypes.bool,
+    isOpen: PropTypes.bool,
+    t: PropTypes.func.isRequired,
+    i18n: PropTypes.any,
+    toolButtonObjects: PropTypes.object.isRequired,
+    openElement: PropTypes.func.isRequired,
+    closeElement: PropTypes.func.isRequired,
+    closeElements: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.overlay = React.createRef();
+
+    const { toolButtonObjects } = props;
+    const dataElement = toolButtonObjects[TOOL_NAME].dataElement;
+    this.state = {
+      left: 0,
+      right: 'auto',
+      top: 0,
+      defaultAnnotations: [],
+      dataElement,
+      language: props.i18n.language,
+    };
+    this.stampTool = core.getTool(TOOL_NAME);
+  }
+
+  componentDidUpdate(prevProps) {
+    // if langauge changes while overlay is open we wanted update it
+    var isLanChanged = this.state.language && this.props.i18n.language !== this.state.language;
+
+    if (!prevProps.isOpen && !this.props.isOpen && this.props.activeToolName === TOOL_NAME) {
```

Comment:
I'm a bit lost here. Can you explain what `!prevProps.isOpen && !this.props.isOpen` does? It seems to me that we want to open this overlay when the previous `activeToolName` isn't `AnnotationCreateRubberStamp` but `this.props.activeToolName` is? If this is the case I think we can just do something like:
```javascript
if (prevProps.activeToolName !== this.props.activeToolName && this.props.activeToolName === TOOL_NAME)
```